Userland/Libraries/LibWeb/HTML/HTMLAudioElement.cpp

```diff
@@ -61,6 +61,8 @@ void HTMLAudioElement::on_seek(double position, MediaSeekMode seek_mode)
     audio_tracks()->for_each_enabled_track([&](auto& audio_track) {
         audio_track.seek(position, seek_mode);
     });
+
+    finish_seek_element(Badge<HTMLAudioElement>());
```

Comment:
This doesn't behave very well on Lagom, where seeking is a bit more asynchronous. The audio runs in its own thread in Ladybird/AudioCodecPluginLadybird.cpp. When I scrub audio with this patch, it's quite jumpy as we are telling `HTMLMediaElement` here that we've finished seeking, but in reality that thread hasn't seeked yet.

I think `HTMLAudioElement` can keep a count of ongoing seeks (incremented in the loop above). Then `Web::Platform::AudioCodecPlugin` can have a callback added (something like `Function<void()> on_seek_complete`, that is invoked once we've actually finished seeking*. In HTMLAudioElement, that callback can decrement the counter, then call `finish_seek_element` when it reaches 0.

\* For `AudioCodecPluginLadybird` you'll have to emit a signal from the audio thread back to the main thread for the main thread to invoke the callback. See how `on_playback_position_updated` is handled for an example.

Reply:
> We basically always create the Badge objects like this:
> 
> ```c++
> finish_seek_element({});
> ```
> 
> That said, I wouldn't worry too much about Badge-protecting this, a public method is fine so we don't have to have one overload per subclass.

The explicit constructor call was because it was overloaded, but I've made it public so that isn't necessary. Thanks!

> This doesn't behave very well on Lagom, where seeking is a bit more asynchronous. The audio runs in its own thread in Ladybird/AudioCodecPluginLadybird.cpp. When I scrub audio with this patch, it's quite jumpy as we are telling `HTMLMediaElement` here that we've finished seeking, but in reality that thread hasn't seeked yet.
> 
> I think `HTMLAudioElement` can keep a count of ongoing seeks (incremented in the loop above). Then `Web::Platform::AudioCodecPlugin` can have a callback added (something like `Function<void()> on_seek_complete`, that is invoked once we've actually finished seeking*. In HTMLAudioElement, that callback can decrement the counter, then call `finish_seek_element` when it reaches 0.
> 
> * For `AudioCodecPluginLadybird` you'll have to emit a signal from the audio thread back to the main thread for the main thread to invoke the callback. See how `on_playback_position_updated` is handled for an example.

I've made these changes, please test to make sure it solves the problems you're seeing, since I unfortunately can't seem to test it properly on my end.